project/services/saveUserService.php

```diff
@@ -66,12 +66,39 @@ function saveUser($email, $name, $surname, $password, $passwordConfirm) {
 
 
 function saveUserWithAdditionalData($userData) {
-    // saveUserAdditionalDataToDB( -- params -- );
-    return null;
+
```

Comment:
I would propose to delete this function as it seems to be same as saveUser();

I would instead propose, to overload functions saveUser() and saveUserToDB() so they take optional parameters 

Example foo($required, $optional = null) {

}

Reply:
ok, that sounds good

Reply:
since CSV service produces like 13 attributes for each user, would it maybe be better to use an associative array to pass the values to the methods instead of distinct parameters?  because a function taking in 13 strings just looks terrible. Whats your take on this?

Reply:
yeah that will come in handy.
what bothers me is saving users from Csv files. They all have 2 different addresses both of which contain several FKs. So its necessary to find or create the associated entries. that's a lot of checking that doesnt normalny happen with manual registration (since the user only inputs few of the parameters)

Reply:
Yes, it will probably be needed for every parameter
- to check if it is in DB
  - if it is, assign that ID to newly created address
  - if not, save it, then retrieve its ID and then assign that ID to newly created address

For every parameter, this can be quite a lot of code. I will probably look at saving Address to DB tomorrow (Tuesday)

Reply:
For now, just send it all to saveUserToDB function, where SQL is handled, and I will try o do something with it from there

Reply:
okay, i plan to create table generating utilities tomorrow and refactor this piece of code as well